extensions/vue/languages/vue/injections.scm

```diff
@@ -1,7 +1,54 @@
-(script_element
-  (raw_text) @content
+; <script lang="js">
+((script_element
+  (start_tag
+    (attribute
+      (attribute_name) @_lang
+      (quoted_attribute_value
+        (attribute_value) @_js)))
+  (raw_text) @content)
+  (#eq? @_lang "lang")
+  (#eq? @_js "js")
   (#set! "language" "javascript"))
 
+; <script lang="ts">
+((script_element
+  (start_tag
+    (attribute
+      (attribute_name) @_lang
+      (quoted_attribute_value
+        (attribute_value) @_ts)))
+  (raw_text) @content)
+  (#eq? @_lang "lang")
+  (#eq? @_ts "ts")
+  (#set! "language" "typescript"))
+
+; <script lang="tsx">
+; <script lang="jsx">
+; Zed built-in tsx, we mark it as tsx ^:)
+(script_element
+  (start_tag
+    (attribute
+      (attribute_name) @_attr
+      (quoted_attribute_value
+        (attribute_value) @tsx_jsx)))
+  (raw_text) @content)
+  (#eq? @_attr "lang")
+  (#any-of? @tsx_jsx "tsx" "jsx")
+  (#set! "language" "tsx"))
+
+
+; {{ }}
+((interpolation
+  (raw_text) @content)
+  (#set! "language" "typescript"))
+
+; v-
+(directive_attribute
+  (quoted_attribute_value
+    (attribute_value) @content
+    (#set! "language" "typescript")))
```

Comment:
This is gonna be great! 🔥 

[**question**] Should these be injected as `javascript` to match the default injection for `<script>`? Or will that  not matter because typescript will also parse/highlight js correctly? (I'm not very familiar with ts.)

Reply:
Yes @claytonrcarter `javascript` should be injected by default for `<script>` without any `lang` attribute.

Reply:
Thanks for the tip, I injected the default <script> -> javascript